core/server/worker/src/main/java/alluxio/worker/block/io/BlockStreamTracker.java

```diff
@@ -21,14 +21,14 @@
  */
 public class BlockStreamTracker {
   /** List of listeners for this tracker. */
-  private static Set<BlockStreamListener> sListeners = new ConcurrentHashSet<>();
+  private static Set<BlockClientListener> sListeners = new ConcurrentHashSet<>();
```

Comment:
Do the `BlockClientListener` implementations implement `hashCode` and `equals`?

Reply:
It doesn't and there is only 1 registrar.
Changed it to a synchronized list anyways.

Reply:
won't adding/removing be slow then? Under heavy load we'd have to iterate over at minimum, the current number of concurrent block requests. With current defaults that's 3k (2k reads, 1k writes)

Reply:
> there is only 1 registrar.

Reply:
and current defaults are not sane.

Reply:
in other words, if this causes an i/o bottleneck then it's a good bottleneck.

Reply:
Changed it to cow list anyway.